File: Simulation/2696.Minimum String Length After Removing Substrings/Solution.java
class Solution {

    Set<String> t = new HashSet(Arrays.asList("AB", "CD"));

    public String dfs(String s) {
        int n = s.length();
        for (int i = 0; i < n - 1; i++) {
            if (t.contains(s.substring(i, i + 2))) {
                return dfs(s.substring(0, i) + s.substring(i + 2));
            }
        }
        return s;
    }

    public int minLength(String s) {
        return dfs(s).length();
    }
}

File: Simulation/2696.Minimum String Length After Removing Substrings/Solution.py
class Solution:
    def minLength(self, s: str) -> int:
        def dfs(s):
            n = len(s)
            for i in range(n):
                if i + 1 < n and s[i:i + 2] in {'AB', 'CD'}:
                    return dfs(s[:i] + s[i + 2:])
            return s

        return len(dfs(s))

